src/cockpit/channels/http_stream2.py

```diff
@@ -0,0 +1,130 @@
+# This file is part of Cockpit.
+#
+# Copyright (C) 2022 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+import http.client
+
+from ..channel import Channel
+
+logger = logging.getLogger(__name__)
+
+
+class HttpStream2Channel(Channel):
+    payload = 'http-stream2'
+    connections = {}
+
+    def get_connection(self, name):
+        if name and name in HttpStream2Channel.connections:
+            logger.debug('reusing connection %s', name)
+            return HttpStream2Channel.connections[name]
+        return None
+
+    def create_connection(self):
+        connection = self.options.get('connection')
+        address = self.options.get('address') or "localhost"
+        port = self.options.get('port')
+        tls = self.options.get('tls')
+        logger.debug('creating new connection for %s to %s:%s; tls: %s', connection, address, port, tls)
+        _cls = http.client.HTTPSConnection if tls is not None else http.client.HTTPConnection
+        con = _cls(address, port)
+        try:
+            # explicitly call connect(), so that we can do proper error handling
+            con.connect()
+        except Exception as e:
```

Comment:
Please try to list the exceptions that you expect might happen here.  This wide net would catch (for example) even the case where `con` was `None` or the `connect()` method was not available.

If it would be totally unwieldy to do so, then at least add a comment about why you need to catch `Exception`.

Reply:
Indeed, that was just a leftover from sloppy initial experiments. Done.